### PR TITLE
add --config-dir support to shellclear cli

### PR DESCRIPTION
**Pull request by boaz-quotient**
I am not sure if this exists or is unnecessary but I couldn't figure out how to set the configuration files directory through the CLI and I want to save it in a different location than `homedir`.
So I decided to add this to the code, let me know if that works out for you.
Thanks.

### CODE REVIEW

File: shellclear/src/config.rs [hunk 1]
```diff
@@ -28,6 +28,17 @@ impl Default for Config {
     }
 }
 
+impl From<Option<&str>> for Config {
+    fn from(config_dir_path_option: Option<&str>) -> Self {
```
Comment (jondot) [inline shellclear/src/config.rs:32]: Naming: you can just use `config_dir` as the parameter name

**Review by jondot — APPROVED** [review]

**Comment by jondot** [comment]
@boaz-quotient I've approved running CI, you can take a look at the linting results for example, which might hints your editor isn't using `fmt` automatically. Would you like help with that?

File: shellclear/src/config.rs (hunk 1 above)
Comment (boaz-quotient) [reply shellclear/src/config.rs]: @jondot thanks for your review.
I updated the variable name and fixed the linting issue by running `cargo fmt` as you suggested.
I wonder if I can auto-format rust code in `vscode` (I installed `rust-analyzer` plugin).

Comment (jondot) [reply shellclear/src/config.rs]: Yes, add the following

Format all languages on save:
```json
  "editor.formatOnSave": true,
```

For rust, use the `rust-analyzer` formatter:

```json
 "[rust]": {
    "editor.defaultFormatter": "rust-lang.rust-analyzer"
  },
```

Comment (boaz-quotient) [reply shellclear/src/config.rs]: I also briefly updated the README file.

Comment (boaz-quotient) [reply shellclear/src/config.rs]: Thanks, rust code is now auto-formatted on `vscode` :laughing:

**Comment by boaz-quotient** [comment]
Sorry for these chaotic updates. I changed the unit tests to use snapshot + added one more unit test (hopefully this will make the coverage happy)

**Comment by jondot** [comment]
No worries, everything looks good to me 👍 
@kaplanelad is leading this project, when he'll be back from vacation he'll probably cut a new release :)

**Comment by boaz-quotient** [comment]
@jondot "ein bha-ya". No worries it's alright. I personally enjoyed practicing Rust on this project. Thanks for the opportunity.

**Comment by kaplanelad** [comment]
Thanks, @boaz-quotientm bumped a new version 0.4.1/

Thanks @jondot for the backup :)

**Pull request merged**